File: Sources/XMTP/Codecs/DecodedComposite.swift
//
//  DecodedComposite.swift
//
//
//  Created by Pat Nakajima on 12/22/22.
//

import Foundation

public struct DecodedComposite {
	var parts: [DecodedComposite] = []
	var encodedContent: EncodedContent?

	init(parts: [DecodedComposite] = [], encodedContent: EncodedContent? = nil) {
		self.parts = parts
		self.encodedContent = encodedContent
	}

	func content<T>() throws -> T? {
		return try encodedContent?.decoded()
	}
}

File: library/src/main/java/org/xmtp/android/library/codecs/DecodedComposite.kt
package org.xmtp.android.library.codecs

data class DecodedComposite(
    var parts: List<DecodedComposite> = listOf(),
    var encodedContent: EncodedContent? = null
) {
    fun <T> content(): T? =
        encodedContent?.decoded()
}
